### PR TITLE
Feature/profile CLI backend

### PR DESCRIPTION
**Pull request by AdaraPutri**
## 📝 Description

This PR adds a standalone user profile feature that lets users save and edit resume-level personal information outside of the resume menu. The goal was to keep profile editing separate from resume editing, since those will live on different frontend pages later.

The profile now stores and lets users update:
- full name (for display)
- email
- phone
- LinkedIn URL
- GitHub URL
- location
- profile paragraph

Resume export now reads from that profile data instead of hardcoded placeholders. If a field is not populated, it is omitted from the exported resume. If the profile paragraph is empty, the whole `Profile` section is hidden. For LinkedIn and GitHub, the exported resume shows clickable `LinkedIn` / `GitHub` labels instead of printing the raw URLs.

Added/updated tests cover:
* standalone user profile defaults for a new user
* saving, updating, and clearing standalone user profile fields
* helper behavior for cleaned contact fields, visible profile text, and full-name fallback
* DOCX export rendering populated profile/contact fields correctly
* DOCX export using full name when present and falling back to username otherwise
* DOCX export rendering LinkedIn/GitHub as hyperlink labels instead of raw visible URLs
* DOCX export hiding contact/profile content when fields are empty
* PDF export rendering the same profile behavior consistently, including full name and hyperlink labels
* PDF export hiding contact/profile content when fields are empty

**Closes:** #568 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [x] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing
* [x] ran `python -m pytest`
* [x] Manually tested by:
- selecting the new `Edit profile` option from the main menu
- entering values for email, phone, LinkedIn, GitHub, location, and profile paragraph
- exporting the same resume to both DOCX and PDF and confirming:
  - populated fields show up
  - empty fields are omitted
  - the `Profile` section is hidden if no profile paragraph is saved
  - LinkedIn and GitHub appear as clickable labels instead of raw links
---

## ✓ Checklist

* [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
* [x] 💬 I have commented my code where needed
* [ ] 📖 I have made corresponding changes to the documentation
* [x] ⚠️ My changes generate no new warnings
* [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
* [ ] 🔗 Any dependent changes have been merged and published in downstream modules
* [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots
Resume if no profile info:
Screenshot 2026-03-10 at 2 57 06 PM
Screenshot 2026-03-10 at 2 57 16 PM

Resume if added profile info:
Screenshot 2026-03-10 at 3 34 19 PM
Screenshot 2026-03-10 at 3 33 59 PM

### CODE REVIEW

**Review by taoTimTim — CHANGES_REQUESTED** [review]
Hi adara, thanks for taking this on.

All tests are passing and the code fits into the system well.

Changing the first profile settings are working great, it's setting them and saving them correctly, but I only entered my email and github. When I went back to then add my phone number and location it made me re-enter my email and github. If I opt to not retype it, choosing "-", it treats the value as NULL. Is it possible to change this so that instead of it being an incremenetal upload (meaning the user must enter everything, every time), the user can choose which one they want to update and *only* that one is updated, without changing the rest of their settings? 

Example, all the options are given, and the user can click 1. for email, 2. for phone, etc.? 

Additionally, is it possible for the user to change their full name, or their username? Meaning, if I make my name "Timmi" in our system, I might actually want "Timandra Draper" on my resume. Thus, username and full_name would be different, and only the username needs to be unique. Can you add this into the `user_profiles` table?

Great work on everything so far.

**Review by taoTimTim — APPROVED** [review]
Thanks for adding the full name, it's properly being used in the resume:
Image

I didn't realize pressing Enter would keep it, so there actually is no problem. Thanks for clarifying that! Approved.

File: tests/test_resume_docx.py [hunk 1]
```diff
@@ -1,246 +1,70 @@
 import json
 from pathlib import Path
+import xml.etree.ElementTree as ET
+import zipfile
```
Comment (ammaarkhan) [inline tests/test_resume_docx.py]: A few of the old tests that got removed here aren't covered by the new ones or anywhere else in the suite:

- Section ordering (PROFILE before SKILLS before PROJECTS before EDUCATION)
- Project sorting by date (newer projects render before older ones)
- Malformed JSON fallback (bad `resume_json` falls back to `rendered_text`)
- Missing `rendered_text` (shows the "data is missing" message)
- Flow edge cases: no saved resumes, cancel/invalid selection, snapshot not found

The new profile tests look good, but it would be worth keeping the old ones alongside them since they cover different things.

**Review by ammaarkhan — COMMENTED** [review]
Hey Adara, I went through the changes and tested locally. The core profile feature looks clean and the hyperlink handling for LinkedIn/GitHub in both DOCX and PDF is a nice touch.

I left a couple of inline comments on the accidental test import in `portfolio_pdf.py` and the unused `pydoc` import in `resume_docx.py`.

On the test rewrites: the new profile tests are solid, but some of the old tests that were removed aren't covered anywhere else in the suite, things like section ordering, project date sorting, malformed JSON fallback, and the flow edge cases. Would be worth bringing those back alongside the new ones since they test different behavior.

Since this is a CLI-focused PR, the API export endpoints not being updated makes sense. Just something to keep in mind for a follow-up so frontend exports pick up profile data too.

Nice work overall!

Before Editing Profile:
Screenshot 2026-03-11 at 6 27 54 PM

After Editing Profile:
Screenshot 2026-03-11 at 7 49 43 PM

**Comment by AdaraPutri** [comment]
@ammaarkhan thanks for the review Ammaar! you're right about the simplified test scripts not covering the important previous test cases. I have reverted the commit that erased the old ones and have kept the newly added ones as well. feel free to re-review! yess the API endpoint refactoring/ additions and frontend implementation for profile will be done by Ivona this week.

**Review by ammaarkhan — APPROVED** [review]
Thanks for adding back the test cases adara and making the other changes. Approved!
Image
